activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -703,6 +713,9 @@ def table_options(table_name)
 
         # strip AUTO_INCREMENT
         raw_table_options.sub(/(ENGINE=\w+)(?: AUTO_INCREMENT=\d+)/, '\1')
```

Comment:
`sub` should be changed to `sub!` as now this is not yet the return value and will be discarded otherwise.